src/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -166,10 +166,11 @@ public static IndexType getIndexType(final BufferedInputStream is) {
      */
     public static Index loadIndex(final String indexFile) {
         final Index idx = null;
+        InputStream inputStream = null;
         BufferedInputStream bufferedInputStream = null;
         final LittleEndianInputStream dis = null;
         try {
-            InputStream  inputStream = ParsingUtils.openInputStream(indexFile);
+            inputStream = ParsingUtils.openInputStream(indexFile);
```

Comment:
actually, reading this again, whatever happens, `inputStream` always ends up wrapped inside of `bufferedInputStream` and that gets a `close()` call, which delegates to the wrapped `inputStream` (and that would close any other stream wrapped inside of `inputStream` - both `BlockCompressedInputStream` and `GZIPInputStream` do that). So the call to `inputStream.close()` is not really needed at all. Am I reading this right?

Also, the `Index` object that gets returned has a stream inside it and that stream is always closed at the beginning. Is that strange?